extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -1012,15 +1014,15 @@ private void possiblyResetOffsetsOrWait(
         // and the current message offset in the kafka partition is more than the
         // next message offset that we are trying to fetch
         if (leastAvailableOffset > nextOffset) {
```

Comment:
Okay, I'm good leaving it as is, since deleting/re-creating a topic is more extreme than messages expiring and I think it'd be appropriate for them to use the reset endpoint manually in that case.